document/core/text/instructions.rst

```diff
@@ -106,6 +115,9 @@ All other control instruction are represented verbatim.
    \production{plain instruction} & \Tplaininstr_I &::=&
      \text{unreachable} &\Rightarrow& \UNREACHABLE \\ &&|&
      \text{nop} &\Rightarrow& \NOP \\ &&|&
+     \text{throw}~~x{:}\Teventidx_I &\Rightarrow& \THROW~x \\ &&|&
+     \text{rethrow} &\Rightarrow& \RETHROW \\ &&|&
+     \text{br\_on\_exn}~~l{:}\Tlabelidx_I~~x{:}\Teventidx_I &\Rightarrow& \BRONEXN~l~x \\ &&|&
```

Comment:
In line with the direction we have taken for other similar cases, the text format syntax for this instruction should probably be
```
br_on_exn $l (exception $x)
```

Reply:
I'm not sure which other cases you are referring to. In this list of plain instruction productions, the other instructions are mixed: function and table indices are written as plain indices, only typeuses are surrounded in "(type ...)". But I agree that exception indices are more like type indices, than function or table indices.
Having said that, I think [wast expects br_on_exn](https://github.com/bytecodealliance/wasm-tools/blob/main/crates/wast/src/ast/expr.rs#L1352) to be followed by two indices, a label and an exception index. Shall I make the change you suggest anyway?

Reply:
Moreover (as I said in my previous comment), if that text format is `br_on_exn l (exception x)`, wouldn't it be confusing that `x` is a `typeidx` instead of an `eventidx`?

Reply:
Yeah, I'm fine addressing this separately, since it requires a change to what's currently implemented.

The idea is that in all cases where an instruction has an index that is not of the "primary" sort of the construct, the text format makes the kind explicit. We do that e.g. for `call_indirect (type $x)`, `data (memory $x)`, `elem (table $x)`, etc., and there will probably be more in the future.

> if that text format is `br_on_exn l (exception x)`, wouldn't it be confusing that `x` is a `typeidx` instead of an `eventidx`?

Not sure I follow, `x` is an eventidx/exnidx, isn't it?

Reply:
> Not sure I follow, x is an eventidx/exnidx, isn't it?

Yes it is an eventidx/exnidx, sorry for not being clear enough. I was also wrong in my assumption that it would be confusing. What I meant was that it would be confusing if an event/exception type were defined with `(exception typeidx)` but that is wrong, it's defined with `(exception (type typeidx))` so please ignore this comment.

I would change this in the next commit as well to what you suggested, and then mention that change at the wast repository. Other implementations would notice the change when tests are added the latest.

Reply:
For some reason this comment doesn't seem to appear at the main discussion, so here's a tag @rossberg, in case you haven't seen this.

Reply:
Thanks, SGTM.